### DISCUSSION TITLE
necessity of self.add_tokens

### DISCUSSION THREAD

**Discussion by cnlinxi** [NVIDIA/NeMo-text-processing#54] [Q&A] [answered]
Thank you for contributing such a valuable project 👍 
I just have some questions:
1. Why use `self.add_tokens`:

for examples, 
```python

graph_numstring = self.add_tokens(pynutil.insert("integer: \"") + graph_numstring + pynutil.insert("\""))
```

in `https://github.com/NVIDIA/NeMo-text-processing/tree/main/nemo_text_processing/text_normalization/zh/taggers/cardinal.py`

2. Why insert `integer: "..."`?

Thank for your reply

**Comment by jimregan** [comment, marked as answer]
`self.add_tokens` adds `self.name` and wraps the content with `{ }`, so in this case you get `cardinal { integer: "..." }`

**Reply by cnlinxi** [reply]
Thank you for your reply. Got it
